File: ryan/week4/special.py
def solution(n):
    snail = [[0] * i for i in range(1, n + 1)]

    x = -1
    y = 0
    num = 1

    for i in range(n):
        for _ in range(i, n):
            # 위에서 왼쪽에서 아래로
            if i % 3 == 0:
                x += 1
            # 아래에서 오른쪽으로
            elif i % 3 == 1:
                y += 1
            # 오른쪽에서 위로
            elif i % 3 == 2:
                y -= 1
                x -= 1

            snail[x][y] = num
            num += 1
    answer = []
    for i in snail:
        answer += i
    return answer

print(solution(4))

File: dongwoo/week4/special.java
import java.util.*;
class Solution {
    public int[] solution(int n) {

        int[] dx = {0, 1, -1};
        int[] dy = {1, 0, -1};

        int[][] temp = new int[n][n];
        int max = n * (n+1) / 2;

        int x = 0;
        int y = 0;
        int num = 1;
        int dir = 0;
        for (int i = 0; i < max; i++) {
            temp[y][x] = num;
            num++;
            int nx = x + dx[dir];
            int ny = y + dy[dir];

            if (nx >= n || ny >= n || nx < 0 || ny < 0 || temp[ny][nx] != 0) {
                dir = (dir + 1) % 3;
                nx = x + dx[dir];
                ny = y + dy[dir];
            }

            x = nx;
            y = ny;
        }

        List<Integer> answer = new ArrayList<>();

        for (int i = 0; i < n; i++) {
            for (int j = 0; j <= i; j++) {
                answer.add(temp[i][j]);
            }
        }
        return answer.stream().mapToInt(i->i).toArray();
    }
}
